### DISCUSSION TITLE
Healthcheck

### DISCUSSION THREAD

**Discussion by kaysond** [thomseddon/traefik-forward-auth#209] [Ideas]
Would it be possible to get a healthcheck added to this? Since the container is built from scratch and its only a single executable, I can't run `wget` or `curl` to ping the listener. Might be easiest to make another simple healthcheck binary that just makes an http request and exits 1 on failure?

**Comment by thomseddon** [comment]
What's the use case for running a health check binary inside the container? Normally this would be performed by the orchestrator?

It could perhaps make sense to add a healthz style endpoint which can be polled, but the binary should exit if there's any major issue, which the orchestrator should pickup and therefore reschedule.

Moving to discussion

**Reply by kaysond** [reply]
At least in Docker, the built in healthcheck system runs commands from within the container, not on the host. Since you've built the container from scratch, there's no wget/curl inside to poll any kind of endpoint.

You're right that the orchestrator would take care of any scenario in which the binary exits, but the point of the healthcheck is to cover scenarios where that doesn't happen for whatever reason.

**Comment by modem7** [comment]
A /healthz style endpoint would certainly be useful for situations where the binary doesn't exit, however rare they might be

**Comment by AnderssonPeter** [comment]
@thomseddon are there any news for adding support for health checks when running this on a plain docker host?
For `Traefik` services I have something like:
```yml
healthcheck:
  test: ['CMD', 'traefik', 'healthcheck', '--ping']
  interval: 10s
  timeout: 10s
  retries: 5
```
They provide a separate command on the binary just for this case that simply calls `http://127.0.0.1/ping`, any chance you could add the same? (If I remember correctly only exit code 0 and 1 should be returned, 0 = success 1 = fail)

**Comment by strausmann** [comment]
@thomseddon is there a timely solution to this issue? There are already some suggestions for a solution. What is the problem with the integration now?

**Reply by nOw-Ay** [reply]
@strausmann please check[ this discussion](https://github.com/thomseddon/traefik-forward-auth/discussions/379), it seems that the current maintainer does not want to actively maintain this repo anymore.
